contracts/interfaces/random-beacon/IRandomBeacon.sol

```diff
@@ -0,0 +1,194 @@
+  error ErrInvalidProof();
+  /// @dev Throws if the request is not finalized
+  error ErrBeaconNotFinalized(uint256 period);
+  /// @dev Throws if the period is invalid (too early or too late)
+  error ErrInvalidPeriod();
+  /// @dev Throws if the request is already submitted
+  error ErrAlreadySubmitted();
+  /// @dev Throws if the request is already finalized
+  error ErrBeaconAlreadyFinalized(uint256 period);
+  /// @dev Throws if the random request is unexists
+  error ErrInvalidRandomRequest(bytes32 expected, bytes32 got);
+  /// @dev Throws if the key hash is not match with the one in the profile
+  error ErrInvalidKeyHash(bytes32 expected, bytes32 actual);
+
+  /**
+   * @dev The validator type.
+   */
+  enum ValidatorType {
+    Unknown,
+    // Max Validator Number
+    All,
+    // Max Governance Validator Number
+    Governance,
+    // Max Standard Validator Number
+    Standard,
+    // Max Rotating Validator Number
+    Rotate
+  }
+
+  /**
+   * @dev The beacon struct.
+   */
+  struct Beacon {
+    // The request hash.
+    bytes32 reqHash;
+    // The random beacon value.
+    uint256 value;
+    // Whether the beacon is finalized.
+    bool finalized;
+    // The submission count.
+    uint32 submissionCount;
+    // Mapping of submitted requests.
+    mapping(address cid => bool) submitted;
+  }
+
+  /**
+   * @dev Emitted when the validator threshold is updated.
+   * @param validatorType The validator type.
+   * @param threshold The new value.
+   */
+  event ValidatorThresholdUpdated(ValidatorType indexed validatorType, uint256 threshold);
+
+  /**
+   * @dev Emitted when the unavailability slash threshold is updated.
+   * @param value The new value.
+   */
+  event SlashUnavailabilityThresholdUpdated(uint256 value);
+
+  /**
+   * @dev Emitted when the beacon is finalized.
+   * @param period The period.
+   * @param value The beacon value.
+   */
+  event BeaconFinalized(uint256 indexed period, uint256 value);
+
+  /**
+   * @dev Emitted when the random seed is fulfilled.
+   * @param by The address that fulfill the random seed.
+   * @param period The period.
+   * @param reqHash The request hash.
+   */
+  event RandomSeedFulfilled(address indexed by, uint256 indexed period, bytes32 indexed reqHash);
+
+  /**
+   * @dev Emitted when the random seed is requested.
+   * @param period The period.
+   * @param reqHash The request hash.
+   * @param req The random request.
+   */
+  event RandomSeedRequested(uint256 indexed period, bytes32 indexed reqHash, RandomRequest req);
+
+  /**
+   * @dev Callback function called at the end of an epoch to wrap up the current epoch and start a new one.
+   * @param lastPeriod The index of the last period in the current epoch.
+   * @param newPeriod The index of the first period in the new epoch.
+   *
+   * if period is ending, finalize the beacon and record the unavailability
+   * if period is not ending and at the start of period, request the random seed for the next period
```

Comment:
Bring these lines to under @dev tag.
**Review other methods' docstring as well!!!**

Reply:
will fix in #29

Reply:
fixed in #29